talaiot/src/main/kotlin/com/cdsap/talaiot/metrics/SimpleMetrics.kt

```diff
@@ -0,0 +1,99 @@
+package com.cdsap.talaiot.metrics
+
+import com.cdsap.talaiot.metrics.base.Metric
+import com.cdsap.talaiot.entities.ExecutionReport
+import com.cdsap.talaiot.metrics.base.GradleMetric
+import org.gradle.api.Project
+import org.gradle.api.internal.GradleInternal
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.internal.scan.scopeids.BuildScanScopeIds
+import oshi.SystemInfo
+import java.util.*
+
+open class SimpleMetric<T>(provider: (Unit) -> T, assigner: (ExecutionReport, T) -> Unit) :
+    Metric<T, Unit>(provider, assigner)
+
+class UserMetric : SimpleMetric<String>(
+    provider = { System.getProperty("user.name") },
+    assigner = { report, value -> report.environment.username = value }
+)
+
+class OsMetric :
+    SimpleMetric<String>(
+        provider = { "${OperatingSystem.current().name}-${OperatingSystem.current().version}" },
+        assigner = { report, value -> report.environment.osVersion = value}
+    )
+
+class BuildIdMetric : SimpleMetric<String>(
+    provider = { UUID.randomUUID().toString() },
+    assigner = { report, value -> report.buildId = value }
+)
+
+/**
+ * As described in the source of gradle:
+ * The ID of a single build invocation.
+ *
+ * Here, the term "build" is used to represent the overall invocation.
+ * For example, buildSrc shares the same build scope ID as the overall build.
+ * All composite participants also share the same build scope ID.
+ * That is, all “nested” builds (in terms of GradleLauncher etc.) share the same build ID.
+ *
+ * This ID is, by definition, not persistent.
+*/
+class GradleBuildInvocationIdMetric : GradleMetric<String>(
+    provider = { project: Project -> (project.gradle as GradleInternal).services[BuildScanScopeIds::class.java].buildInvocationId },
```

Comment:
all the build registered with `Talaiot` will contain a unique ID. In terms of cardinality constraints, can represent a problem?

Reply:
This one is not included by default. There are 2 actually: BuildIdMetric and GradleBuildInvocationIdMetric. The difference is that if you call a project with another execution combined (gradle task wrapper for example) the BuildIdMetric will be different for them.

And BuildIdMetric is under the flag generateBuildId in MetricsConfiguration, off by default